### ISSUE TITLE
Users unable to edit the report they created

### ISSUE THREAD

**Issue by 1yuv** [medic/cht-core#8589]
**Describe the bug**
We've seen CHT user(s) are able to create report but they are not able to edit the same report. 

**To Reproduce**
1. Create a report with offline CHT user for particular contact
2. Go to Reports tab and select same report
3. Click on Edit to edit this report.

**Expected behavior**
Users should be able to edit the report that they created if permission allows that. 

**Screenshots**
See the video:

https://github.com/medic/cht-core/assets/6102813/8c7c7652-efb4-4903-b40b-442438ca53ba

**Environment**
- Instance: walling.dev
- Browser: Chrome
- Client platform: MacOS
- App: Webapp
- Version: 4.3.1 and 4.4.0 both tested

**Additional context**
app settings and forms used for this: medic/config-nssd#406 (internal)

**Comment by dianabarsan** [comment]
@yrimal 
Can you please also share the permissions of the form?

**Comment by yrimal** [comment]
Hi @dianabarsan , 
Here's the forms.properties section 
```json
{
  "title": [
    {
      "locale": "en",
      "content": "Non-communicable diseases survey form"
    },
    {
      "locale": "ne",
      "content": "नसर्ने रोग सम्बन्धी सर्बेक्षण फारम"
    }
  ],
  "context": {
    "place": false,
    "person": true,
    "expression": "contact.contact_type === 'c82_person' && user._id != contact._id"
  },
  "icon": "icon-healthcare-assessment"
}

```

Contact type for which this report was added is `c82_person`.

**Comment by dianabarsan** [comment]
I am able to replicate this with default config and submitting a `death_report` form. It looks like the form expression context does not get populated with current form data when it's evaluated for an edit.

**Comment by dianabarsan** [comment]
I believe this started manifesting when we merged https://github.com/medic/cht-core/issues/6505 and started evaluating form context on edit. The issue is that the context that we sent on edit does not have the same format as on create.

**Comment by garethbowen** [comment]
I'm not 100% convinced by your examples, because in both cases someone is editing a report that is no longer relevant anyway. Why edit a report for a muted contact? Or for a condition that no longer applies to the patient? A particularly fastidious health worker may want to fix the data regardless but I think it's unlikely.

One more concerning example would be if they submitted a report that was incorrect which _caused_ the report itself to be uneditable. For example, if a pregnant woman had a checkup, and one of the questions was "has the baby been delivered?". Answering "yes" might remove the "pregnant" condition which could make the "checkup" form irrelevant, and therefore you couldn't edit the report and turn the "yes" to a "no".

So I think your approach makes sense - for editing check permissions but don't check the expression.

**Comment by dianabarsan** [comment]
Backported fix to `4.3.x`

**Comment by dianabarsan** [comment]
Backported to `4.4.x` and merged to `master`.
